Upgrade to Rails 6

eoinkelly opened the pull request:
Teach the old dog a new trick :smile:

eoinkelly committed:
Basic Rails 6 upgrade

1. Run 'bundle exec rails app:update'
2. Fix issue with ckeditor_rails gem not having an official Rails 6 version
3. Move gems out of the long deprecated 'assets' group in Gemfile

eoinkelly committed:
Adopt Rails 6 config

This change also removes the unused Rails 5.2 framework defaults initializer
as this is no longer required

eoinkelly committed:
Allow force_ssl to be turned off in production env via env var

It defaults on but can be disabled by an env var. This allows us to
start the app in production mode locally to test assets pre-compilation etc.

This change also documents how to start the app in production env locally.

eoinkelly committed:
Upgrade asset pre-compilation instructions and remove now unused assets

eoinkelly committed:
Remove rails_12factor gem which is not required for Rails 5+

See https://github.com/heroku/rails_12factor#rails-5 for details

eoinkelly committed:
Silence deprecation warning

eoinkelly committed:
Allow rubocop -a to fixup changes

eoinkelly committed:
Disable brakeman in Codeclimate

It seems that Codeclimate can't run Brakeman anymore because its license
was changed as part of an aquisition - see:

* https://docs.codeclimate.com/docs/brakeman
* https://brakemanscanner.org/blog/2018/06/28/brakeman-has-been-acquired-by-synopsys/index.html

Since we are already running `brakeman` as part of our CI run, we can
safely disable the outdated version from Codeclimate.

Br3nda commented on Gemfile:
```diff
@@ -53,11 +62,9 @@ gem 'devise'
-  gem 'sass-rails'
-  gem 'uglifier'
-end
+gem 'coffee-rails'
```
we can get rid of this as soon as we fix up `./app/assets/javascripts/ckeditor/config.js.coffee`

eoinkelly replied:
Done

eoinkelly committed:
Remove last remaining coffeescript in the project

eoinkelly committed:
Run eslint --fix to fix JS formatting issues

eoinkelly committed:
Run ESLint as part of CI

eoinkelly committed:
Disable ESLint in Codeclimate because we now run it in CI

Running ESLint in CI is preferable because:

1. CI checks can be run locally so devs can get early feedback about issues
2. We have one consistent version of ESlint

eoinkelly committed:
Delay the call to window.print() to make it more reliable

We are seeing some issues in some browsers where the print dialog is not
appearing. I haven't been able to replicate it locally but I can
replicate it very occasionally on a Heroku review app.

Adding 1 second delay hoping that will even out these issues.

eoinkelly committed:
Merge branch 'staging' into task/rails-6-upgrade
